src/neuroconv/datainterfaces/ophys/femtonics/femtonicsdatainterface.py

```
@@ -0,0 +1,259 @@
+        channel_name: Optional[str] = None,
+        verbose: bool = False,
+    ):
+        """
+        Initialize the FemtonicsImagingInterface.
+
+        Parameters
+        ----------
+        file_path : str or Path
+            Path to the .mesc file.
+        session_index : int, optional
+            Index of the MSession to use (0-based). Default is 0.
+        munit_index : int, optional
+            Index of the MUnit within the specified session (0-based). Default is 0.
+
+            In Femtonics MESc files, an MUnit ("Measurement Unit") represents a single imaging acquisition or experiment,
+            including all associated imaging data and metadata. A single MSession can contain multiple MUnits,
+            each corresponding to a separate imaging run/experiment performed during the session.
+            MUnits are indexed (0, 1, 2, ...) within each session.
+
+        channel_name : str, optional
+            Name of the channel to extract (e.g., 'UG', 'UR').
+            If multiple channels are available and no channel is specified, an error will be raised.
+            If only one channel is available, it will be used automatically.
+        verbose : bool, optional
+            Whether to print verbose output. Default is False.
+        """
+
+        super().__init__(
+            file_path=file_path,
+            session_index=session_index,
+            munit_index=munit_index,
+            channel_name=channel_name,
+            verbose=verbose,
+        )
+
+        self._file_path = file_path
+        self._session_index = session_index
+        self._munit_index = munit_index
+        self._channel_name = channel_name
+
+        # Hack till roiextractors removes the get_num_channels method in `check_imaging_equal`.
+        # TODO: remove this once roiextractors 0.6.1
+        self.imaging_extractor.get_num_channels = lambda: 1  # Override to ensure only one channel is reported
+
+        Extractor = self.get_extractor()
```

We don't have this behavior in the roi extractor, right? We should probably do the same in roiextractors.

---

By this behavior I mean that the arguments are None by default and then extract the first element if they are not specified.

---

You're right, the ROI extractor doesn’t currently follow this behavior. In the FemtonicsImagingExtractor:

- session_index=None: defaults to 0 (i.e., MSession_0) if not set.
- munit_index=None: defaults to 0 (i.e., MUnit_0) if not set.
- channel_name=None:
If only one channel exists, it's used automatically.
If multiple channels exist, an error is raised — user must specify.

Should I open a new issue and PR for the extractor to have the same logic, instead of fixing it in the Interface ?

---

I think you can open an issue or a PR in roiextractors. This we can leave like this at the moment so we can merge. Leave a TODO note so we can fix it here as well once roiextractors next release is done. What do you think?

---

Yes, that sounds good. I'll open an issue or PR in roiextractors and leave a TODO note here to update it after the next release.

---

Sorry, I misread above. 

The behavior for `munit_index` and `session_index` should be exactly as for `channel_name`. If there is only one and the passed value is None (not specified) then the first (and only) found value should be picked. Otherwise, it should raise an informative error telling the user what are the options.

I just realize that the test examples are multi-unit so we need to specify the m_unit_index in most of the examples contrary to what I have been requesting. My apologies .

---

Also, I am thinking that to specify the m_unit we should it by name and not by index. After all, our static method is get_available_units and I think it is better to make the API based on format based concept (the name of the units) rather than programming ones (the index).

---

Yes, exactly. We should do that.

---

done, applied the same logic to the session handling and opened a corresponding [issue](url) in roiextractor to fix the extractor.